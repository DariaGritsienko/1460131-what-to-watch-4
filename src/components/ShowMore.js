import React from 'react';
import {showMoreAction} from '../reducer';
import PropTypes from 'prop-types';

export default function ShowMore({store, onButtonClickMore, isButtonShowMoreActive}) {

  const onClickMore = () => {
    store.dispatch(showMoreAction());
    onButtonClickMore(store.getState().page);
  };

  if (!store || isButtonShowMoreActive || store.getState().films.length <= store.getState().pageSize * (store.getState().page + 1)) {
    return null;
  }

  return (
    <button
      className="catalog__button"
      type="button"
      onClick={() => onClickMore()}
    >
      Show more
    </button>
  );
}

ShowMore.propTypes = {
  isButtonShowMoreActive: PropTypes.bool,
  store: PropTypes.object,
  onButtonClickMore: PropTypes.func
};
