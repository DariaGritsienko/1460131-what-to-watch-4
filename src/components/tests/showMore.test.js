import React from "react";
import renderer from "react-test-renderer";
import ShowMore from '../ShowMore';


describe(`Foo`, () => {
  it(`<ShowMore /> should render ShowMore`, () => {
    const onButtonClickMore = jest.fn();
    const isButtonShowMoreActive = false;
    const tree = renderer.create(
        <ShowMore
          onButtonClickMore={onButtonClickMore}
          isButtonShowMoreActive={isButtonShowMoreActive}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

