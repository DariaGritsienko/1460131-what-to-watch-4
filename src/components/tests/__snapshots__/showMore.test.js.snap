// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foo <ShowMore /> should render ShowMore 1`] = `null`;
